### ISSUE TITLE
Can't connect to diagnostic.opendns.com

### ISSUE THREAD

**Issue by joshrickert** [roots/trellis#434]
I haven't managed to get the provisioner to run today. It keeps failing with this error:

```
curl: (35) Server aborted the SSL handshake
fatal: [default] => Failed to template {{ lookup('pipe', 'curl -4 https://diagnostic.opendns.com/myip') }}: lookup_plugin.pipe(curl -4 https://diagnostic.opendns.com/myip) returned 35
```

I thought it might be an SSL issue, but switching to HTTP just gives me this instead:

```
curl: (52) Empty reply from server
fatal: [default] => Failed to template {{ lookup('pipe', 'curl -4 http://diagnostic.opendns.com/myip') }}: lookup_plugin.pipe(curl -4 http://diagnostic.opendns.com/myip) returned 52
```

These errors seem to inject themselves randomly throughout the provisioning script.

Should I assume it's just a connectivity issue to diagnostic.opendns.com? Any way we can work in a [fallback](http://stackoverflow.com/questions/3097589/getting-my-public-ip-via-api)?

**Comment by QWp6t** [comment]
Does this work?: `{{ lookup('pipe', 'dig +short myip.opendns.com @resolver1.opendns.com') }}`

That's what we used to have, but some networks disallow UDP connections to external DNS servers, which caused it to fail.

Give it a shot to see if it works for you.

Otherwise, you can certainly use any of the others mentioned on the site you linked; this worked for me, for example: `{{ lookup('pipe', 'curl -4 https://ipinfo.io/ip') }}`.

Instead of coding in a fallback, we could probably give users the ability specify their own external IP lookup mechanism. @fullyint @swalkinshaw thoughts?

**Comment by swalkinshaw** [comment]
@QWp6t I'm not opposed to having it customizable as long as it's simple.

**Comment by joshrickert** [comment]
I gave it another shot today (with the default config) and Ansible ran successfully. It's probably an intermittent connectivity issue, either on the provider's end or mine.

I couldn't get any of the alternative services I mentioned working consistently either yesterday. If I spammed the `curl` command into my console I saw failures in maybe 1/10 cases.

I'll close this for now, but at least it's documented in case anyone else has issues in the future.
